wasm-node/src/node.rs

```diff
@@ -0,0 +1,179 @@
+use anyhow::Context;
+use celestia_node::network::network_id;
+use celestia_node::node::{Node, NodeConfig};
+use celestia_node::store::IndexedDbStore;
+use celestia_types::hash::Hash;
+use js_sys::{Array, JsString};
+use libp2p::identity::Keypair;
+use libp2p::{identity, Multiaddr};
+use serde_wasm_bindgen::to_value;
+use tracing::info;
+use wasm_bindgen::prelude::*;
+
+use crate::utils::Network;
+use crate::wrapper::libp2p::NetworkInfo;
+
+#[wasm_bindgen(js_name = Node)]
+struct WasmNode {
+    node: Node<IndexedDbStore>,
+}
+
+#[wasm_bindgen(js_name = NodeConfig)]
+pub struct WasmNodeConfig {
+    pub network: Network,
+    #[wasm_bindgen(skip)]
+    pub genesis_hash: Option<Hash>,
+    #[wasm_bindgen(skip)]
+    pub p2p_local_keypair: Keypair,
+    #[wasm_bindgen(skip)]
+    pub p2p_bootnodes: Vec<Multiaddr>,
```

Comment:
wouldn't `Keypair` and `Multiaddr` be supported with some `wasm` feature in libp2p?
If so, then we could use the `[u8; HASH_SIZE]` for hash and remove the wasm_bindgen stuff.

We can still translate to and from some types in `WasmNode::new`

Reply:
Unfortunately, no. Multiaddr comes from `multiaddr` crate and isn't wasm-bindgen aware. `Keypair` comes from `libp2p` but also isn't.